### PR TITLE
Improve Threading & Ditch Long Polling

### PR DESCRIPTION
**Pull request by gmbnt**

### CODE REVIEW

File: src/org/thoughtcrime/securesms/loki/redesign/messaging/LokiPublicChatPoller.kt [hunk 1]
```diff
@@ -235,22 +236,24 @@ class LokiPublicChatPoller(private val context: Context, private val group: Loki
...
-        }.successBackground {
+        }.success {
             val newDisplayNameUpdatees = uniqueDevices.mapNotNull {
                 // This will return null if the current device is a master device
                 LokiDeviceLinkUtilities.getMasterHexEncodedPublicKey(it).get()
             }.toSet()
             // Fetch the display names of the master devices
             displayNameUpdatees = displayNameUpdatees.union(newDisplayNameUpdatees)
-        }.successBackground { messages ->
+        }.success { messages ->
```
Comment (Mikunj) [inline src/org/thoughtcrime/securesms/loki/redesign/messaging/LokiPublicChatPoller.kt]: `successBackground` does the same thing as what you just did below

File: src/org/thoughtcrime/securesms/loki/redesign/messaging/LokiPublicChatPoller.kt [hunk 2]
```diff
@@ -260,26 +263,30 @@ class LokiPublicChatPoller(private val context: Context, private val group: Loki
         if (displayNameUpdatees.isEmpty()) { return }
         val hexEncodedPublicKeys = displayNameUpdatees
         displayNameUpdatees = setOf()
-        api.getDisplayNames(hexEncodedPublicKeys, group.server).successBackground { mapping ->
-            for (pair in mapping.entries) {
-                val senderDisplayName = "${pair.value} (...${pair.key.takeLast(8)})"
-                DatabaseFactory.getLokiUserDatabase(context).setServerDisplayName(group.id, pair.key, senderDisplayName)
-            }
+        api.getDisplayNames(hexEncodedPublicKeys, group.server).success { mapping ->
```
Comment (Mikunj) [inline src/org/thoughtcrime/securesms/loki/redesign/messaging/LokiPublicChatPoller.kt]: Same thing here, `successBackground` just does whatever you did

File: src/org/thoughtcrime/securesms/loki/redesign/messaging/LokiPublicChatPoller.kt (hunk 1 above)
Comment (gmbnt) [reply src/org/thoughtcrime/securesms/loki/redesign/messaging/LokiPublicChatPoller.kt]: I know. Was trying to make it more explicit what's going on.

File: src/org/thoughtcrime/securesms/loki/redesign/messaging/LokiPublicChatPoller.kt (hunk 2 above)
Comment (gmbnt) [reply src/org/thoughtcrime/securesms/loki/redesign/messaging/LokiPublicChatPoller.kt]: Like I said I feel it's a lot more obvious this way

Comment (gmbnt) [reply src/org/thoughtcrime/securesms/loki/redesign/messaging/LokiPublicChatPoller.kt]: We can chat about this and potentially move back to the `successBackground` thing if you feel strongly about it but I'm merging this first to get rid of long polling

**Pull request merged**